internal/manifests/collector/config_replace.go

```diff
@@ -42,18 +42,22 @@ type Config struct {
 	TargetAllocConfig *targetAllocator   `yaml:"target_allocator,omitempty"`
 }
 
-func ReplaceConfig(instance v1alpha1.OpenTelemetryCollector) (string, error) {
+func ReplaceConfig(instance v1alpha2.OpenTelemetryCollector) (string, error) {
+	cfgStr, err := instance.Spec.Config.Yaml()
+	if err != nil {
+		return "", err
+	}
 	// Check if TargetAllocator is enabled, if not, return the original config
 	if !instance.Spec.TargetAllocator.Enabled {
-		return instance.Spec.Config, nil
+		return string(cfgStr), nil
```

Comment:
If the `Yaml` method returns a string, why do you need to convert it to `string`?